### PR TITLE
feat: Kubectl proxy support

### PR DESCRIPTION
**Pull request by kartik-579**
# Description

Updated ClusterForm component to handle KubectlProxy Checkbox.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/cluster/ClusterForm.tsx [hunk 1]
```diff
@@ -123,6 +126,7 @@ export default function ClusterForm({
     const [selectAll, setSelectAll] = useState<boolean>(false)
     const [getClusterVar, setGetClusterState] = useState<boolean>(false)
     const [isVirtual, setIsVirtual] = useState(isVirtualCluster)
+    const [toConnectViaProxyTemp,setToConnectViaProxyTemp] = useState(toConnectViaProxy)
```
Comment (shivani170) [inline src/components/cluster/ClusterForm.tsx:129]: format this line

File: src/components/cluster/ClusterForm.tsx [hunk 2]
```diff
@@ -181,6 +186,10 @@ export default function ClusterForm({
             prometheusTlsClientCert: {
                 required: false,
             },
+            proxyUrl: {
+                required: id && KubectlProxyCheckBox ? toConnectViaProxyTemp : false,
```
Comment (shivani170) [inline src/components/cluster/ClusterForm.tsx:190]: (id && KubectlProxyCheckBox) && toConnectViaProxyTemp

File: src/components/cluster/ClusterForm.tsx [hunk 3]
```diff
@@ -598,6 +633,14 @@ export default function ClusterForm({
                         </label>
                     )}
                 </div>
+                {id !== 1 && KubectlProxyCheckBox && (
```
Comment (shivani170) [inline src/components/cluster/ClusterForm.tsx:636]: make 1 as a valid constant

File: src/components/cluster/Cluster.tsx [hunk 4]
```diff
@@ -52,6 +52,7 @@ export default class ClusterList extends Component<ClusterListProps, any> {
             browseFile: false,
             isClusterDetails: false,
             showEditCluster: false,
+            toConnectViaProxy: false,
```
Comment (shivani170) [inline src/components/cluster/Cluster.tsx:55]: Please rename it to isConnectedViaProxy

File: src/components/cluster/Cluster.tsx [hunk 5]
```diff
@@ -325,6 +329,8 @@ function Cluster({
             password: { value: prometheusAuth?.password, error: '' },
             prometheusTlsClientKey: { value: prometheusAuth?.tlsClientKey, error: '' },
             prometheusTlsClientCert: { value: prometheusAuth?.tlsClientCert, error: '' },
+            proxyUrl: {value: proxyUrl, error: '' },
+            toConnectViaProxy: {value: proxyUrl ? proxyUrl!='' : false,error:''},
```
Comment (shivani170) [inline src/components/cluster/Cluster.tsx:333]: It is a boolean please fix line 333

File: src/components/cluster/Cluster.tsx [hunk 6]
```diff
@@ -346,6 +352,13 @@ function Cluster({
                 required: true,
                 validator: { error: 'URL is required', regex: /^.*$/ },
             },
+            proxyUrl: {
+                required: false,
+                validator: { error: 'Incorrect Url', regex: /^.*$/ },
+            },
+            toConnectViaProxy: {
```
Comment (shivani170) [inline src/components/cluster/Cluster.tsx:359]: Is this needed?

File: src/components/cluster/Cluster.tsx [hunk 7]
```diff
@@ -757,6 +777,8 @@ function Cluster({
                                 defaultClusterComponent={state.defaultClusterComponent}
                                 isTlsConnection={isTlsConnection}
                                 isClusterDetails={state.isClusterDetails}
+                                proxyUrl={state.proxyUrl}
+                                toConnectViaProxy={state.toConnectViaProxy.value}
```
Comment (shivani170) [inline src/components/cluster/Cluster.tsx:781]: it must be `state.toConnectViaProxy`

File: src/components/cluster/ClusterForm.tsx (hunk 2 above)
Comment (shivani170) [reply src/components/cluster/ClusterForm.tsx:190]: Are you expecting proxyUrl as a boolean ?

File: src/components/cluster/Cluster.tsx (hunk 6 above)
Comment (kartik-579) [reply src/components/cluster/Cluster.tsx:363]: if this is not present, handleOnChange validator gives error as it is not able to get the key present in the object. we do not need the validation but we need the key to be present for the validator logic to not break

File: src/components/cluster/Cluster.tsx (hunk 4 above)
Comment (kartik-579) [reply src/components/cluster/Cluster.tsx:55]: done

File: src/components/cluster/ClusterForm.tsx (hunk 3 above)
Comment (kartik-579) [reply src/components/cluster/ClusterForm.tsx:636]: done

File: src/components/cluster/ClusterForm.tsx (hunk 2 above)
Comment (kartik-579) [reply src/components/cluster/ClusterForm.tsx:190]: done

File: src/components/cluster/ClusterForm.tsx (hunk 1 above)
Comment (kartik-579) [reply src/components/cluster/ClusterForm.tsx:129]: done

File: src/components/cluster/Cluster.tsx (hunk 7 above)
Comment (kartik-579) [reply src/components/cluster/Cluster.tsx]: done

File: src/components/cluster/Cluster.tsx (hunk 5 above)
Comment (kartik-579) [reply src/components/cluster/Cluster.tsx]: done

**Review by shivani170 — APPROVED** [review]

**Pull request merged**